### PR TITLE
Bug fix. Metrics.Type value was not setting any Enabled flag

### PR DESCRIPTION
**Pull request by guscarreon**
To preserve backwards compatibility, the `Type` field of `config.Metrics` can keep carrying the `"influx"` or `"none"` values to activate metrics without explicitly setting the `Influx` object's `Enabled` boolean field to true. This pull request will translate an `"influx"` or `"prometheus"` value found in `Type`, to an `Enabled` flag set to true. In turn, this change corrects the bug that kept version 0.10.0's influx metrics from running.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -145,23 +145,22 @@ type Metrics struct {
 
 func (cfg *Metrics) validateAndLog() {
 
-	metricsEnabled := false
-
 	if cfg.Type == MetricsInflux || cfg.Influx.Enabled {
 		cfg.Influx.validateAndLog()
-		metricsEnabled = true
+		cfg.Influx.Enabled = true
```
Comment (SyntaxNode) [inline config/config.go:150]: If validation fails, does the `log.Fatalf` call in `validateAndLog` cause the program to exit?

File: config/config.go [hunk 2]
```diff
@@ -178,8 +177,9 @@ func (cfg *Metrics) validateAndLog() {
 type MetricsType string
 
 const (
-	MetricsNone   MetricsType = "none"
-	MetricsInflux MetricsType = "influx"
+	MetricsNone       MetricsType = "none"
+	MetricsInflux     MetricsType = "influx"
+	MetricsPrometheus MetricsType = "prometheus"
```
Comment (SyntaxNode) [inline config/config.go:182]: There's no need to add a MetricsPrometheus IMHO. It's a legacy config we're moving away from.

File: config/config_test.go [hunk 3]
```diff
@@ -373,6 +405,163 @@ func TestCheckMetricsEnabled(t *testing.T) {
...
+			out:         testOut{false, true},
+		},
+		{
+			description: "[12] metricType = \"prometheus\"; Both flags enabled.",
+			in:          testIn{"prometheus", true, true},
+			out:         testOut{true, true},
+		},
+		{
+			description: "[13] metricType = \"trendalyze\"; No flags enabled. ",
+			in:          testIn{"trendalyze", false, false},
```
Comment (SyntaxNode) [inline config/config_test.go:489]: Consider renaming `trendalyze` to `unknown` for clarity in understanding the test values.

File: config/config.go (hunk 1 above)
Comment (guscarreon) [reply config/config.go:150]: Yes it does. Do you want me to change to `log.Infof` and continue with disabled Influx metrics instead?

File: config/config.go (hunk 2 above)
Comment (guscarreon) [reply config/config.go:182]: Makes sense. Removed.

File: config/config_test.go (hunk 3 above)
Comment (guscarreon) [reply config/config_test.go:489]: Done

File: config/config.go (hunk 1 above)
Comment (SyntaxNode) [reply config/config.go:150]: Nope. An exit here is desired.

**Pull request merged**